### PR TITLE
Add `WebSockets` support to `VertxFutureServerInterpreter`

### PR DESCRIPTION
**Pull request by tdroxler**
Hi everyone,

this PR is adding web sockets to the `VertxFutureServerInterpreter`.
 
You'll see that it misses some implementations of some configuration stuff: `ignorePong`, `autoPing`and `contatenateFrames`. 
But this is a first step and the web socket is already working, I could test it in my project.

### CODE REVIEW

File: server/vertx-server/src/main/scala/sttp/tapir/server/vertx/streams/package.scala [hunk 1]
```diff
@@ -21,7 +25,68 @@ package object streams {
...
+  private def optionallyAutoPing(rs:ReadStream[WebSocketFrame], autoPing: Option[(FiniteDuration, WebSocketFrame.Ping)]): ReadStream[WebSocketFrame] = {
+    //TODO implement this
+    rs
+  }
+}
+
+/*
+ * ReadStream doesn't offer a `map` function
+ */
+class ReadStreamMapping[A, B](source: ReadStream[A], mapping: A => B) extends ReadStream[B] {
```
Comment (tdroxler) [inline server/vertx-server/src/main/scala/sttp/tapir/server/vertx/streams/package.scala:64]: unfortunately no `map` function in `ReadStream`  eclipse-vertx/vert.x#3081

**Comment by adamw** [comment]
Thanks - this looks like a good start. One technical remark: I think the code should be reformatted using scalafmt, and secondly - the vertx module stopped compiling using Scala 3. Probably a simple fix somewhere :)

**Comment by tdroxler** [comment]
> Thanks - this looks like a good start. One technical remark: I think the code should be reformatted using scalafmt, and secondly - the vertx module stopped compiling using Scala 3. Probably a simple fix somewhere :)

 Thx for the review, fixes done.

**Comment by adamw** [comment]
Thanks!

**Pull request merged**